lxd/storage/backend_lxd_patches.go

```diff
@@ -1,17 +1,96 @@
 package storage
 
 import (
+	"github.com/pkg/errors"
+
+	"github.com/lxc/lxd/lxd/db"
+	"github.com/lxc/lxd/lxd/project"
+	"github.com/lxc/lxd/lxd/revert"
 	"github.com/lxc/lxd/lxd/storage/drivers"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
+	"github.com/lxc/lxd/shared/logger"
 )
 
 var lxdEarlyPatches = map[string]func(b *lxdBackend) error{}
 
 var lxdLatePatches = map[string]func(b *lxdBackend) error{
-	"storage_create_vm":       lxdPatchStorageCreateVM,
-	"storage_create_vm_again": lxdPatchStorageCreateVM,
+	"storage_create_vm":                        lxdPatchStorageCreateVM,
+	"storage_create_vm_again":                  lxdPatchStorageCreateVM,
+	"storage_rename_custom_volume_add_project": lxdPatchStorageRenameCustomVolumeAddProject,
 }
 
 // Patches start here.
 func lxdPatchStorageCreateVM(b *lxdBackend) error {
 	return b.createStorageStructure(drivers.GetPoolMountPath(b.name))
 }
+
+// lxdPatchStorageRenameCustomVolumeAddProject renames all custom volumes in the default project (which is all of
+// the custom volumes right now) to have the project prefix added to the storage device volume name.
+// This is so we can added project support to custom volumes and avoid any name collisions.
+func lxdPatchStorageRenameCustomVolumeAddProject(b *lxdBackend) error {
+	// At this time, all custom volumes are in the default project.
+	volumes, err := b.state.Cluster.StoragePoolVolumesGet(project.Default, b.ID(), []int{db.StoragePoolVolumeTypeCustom})
```

Comment:
Can you confirm that this only retrieves the ones local to this node and not everything from the cluster?

Reply:
@stgraber ah good spot, have replaced with a call to `StoragePoolNodeVolumesGet` which just gets the node volumes.